### PR TITLE
Players redirected to lobby after end of game

### PR DESCRIPTION
**Pull request by sibela46**

### CODE REVIEW

File: server/proxy/server.js [hunk 1]
```diff
@@ -143,6 +143,22 @@ app.get('/playCard', (req, res) => {
     .catch((err) =>  res.send(err));
 });
 
+app.get('/voteCard', (req, res) => {
```
Comment (jburton-scottlogic) [inline server/proxy/server.js:146]: Struggling to understand what this is doing? Is there a danger of the user being able to pass in their own url here that theyre hosting? or would that even matter?

File: server/services/socket.js [hunk 2]
```diff
@@ -91,6 +91,13 @@ exports.emitWinner = (roomId, player) => io.to(`room-${roomId}`).emit("winner",
 // When game is over, tell the users
 exports.emitEndGame = (roomId) => io.to(`room-${roomId}`).emit("end");
 
+// Close the room when the game has ended
+exports.closeRoom = (roomId) => {
+    sockets.forEach(socket => socket.handshake.session.roomId === roomId && socket.leave(`room-${roomId}`));
```
Comment (jburton-scottlogic) [inline server/services/socket.js:101]: lets say you had 100 rooms and the first one was the correct roomId, you would still be looping around the other 99 here. You could use 'find' to get the first instance in the array that matches.
```

const socket = sockets.find(socket => socket.handshake.session.roomId === roomId);
socket .leave(`room-${roomId}`));

```
or something similar

Comment (jburton-scottlogic) [reply server/services/socket.js:101]: ah sorry i think i misunderstood, there will be multiple sockets that match the room id, ignore this

File: server/proxy/server.js (hunk 1 above)
Comment (sibela46) [reply server/proxy/server.js:146]: I'm not really sure what you mean... this proxy server is ran only during testing mode though

File: server/services/socket.js (hunk 2 above)
Comment (sibela46) [reply server/services/socket.js:101]: Yeah, I think we're looping through the sockets here to make sure they leave their rooms when the game ends

File: server/proxy/server.js (hunk 1 above)
Comment (jburton-scottlogic) [reply server/proxy/server.js:146]: ahhh! got you, nvm!

**Review by jburton-scottlogic — DISMISSED** [review]

**Review by Ro4052 — APPROVED** [review]

**Review by tillywoodfield — APPROVED** [review]

**Pull request merged**